### PR TITLE
 Block Library: Add the Comments Pagination block

### LINKED ISSUE #35007: Post Comments Pagination block
## Description

Displays a paginated navigation to next/previous set of comments, when applicable. This should be similar to the [Query Pagination block](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/query-pagination).

This block is part of the Post Comments Loop block, for more context please visit this [Tracking Issue](https://github.com/WordPress/gutenberg/issues/34994) or its [GitHub project](https://github.com/WordPress/gutenberg/projects/61).

## Potential Settings

- **Color settings:** This should be similar to the [Query Pagination block](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/query-pagination).
- **Pagination Arrow:** It allows users to select if they want none/Arrow/Chevron in the Next & Previous buttons. This should be similar to the [Query Pagination block](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/query-pagination).

## Requirements

- This block should only be available inside the [Post Comments Query Loop Block](https://github.com/WordPress/gutenberg/issues/34996).
- If there is just one page, we shouldn't show the pagination.

## References

This should be similar to the [Query Pagination block](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/query-pagination).

### PR DESCRIPTION
**Pull request by cbravobernal**
## Description

This block works as a container for other pagination blocks that render links to navigate through the list of comments. It basically works the same as the [Query Pagination block](packages/block-library/src/query-pagination), but for comments.

It has the same settings:
- changing layout and colors
- choosing different styles for next/prev arrows. 

Will close #35007
Is a follow-up of #36577

## How has this been tested?

1. Modify the Single Post template, adding a Comments Query Loop block
2. Add inside the Comments Pagination block
3. Check that the layout and color settings work (tested only in the editor)
4. Check that the arrow setting works as expected (tested only in the editor)

## Types of changes
- New feature

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

**Comment by DAreRodz** [comment]
Thanks @c4rl0sbr4v0 for taking care of this.

As I was the author of the original PR, I guess it wouldn't be OK to approve and merge my own PR 😅 , although from what @gziolo said in https://github.com/WordPress/gutenberg/pull/36577#issuecomment-972767776 the PR should be ready to go.

**Comment by gziolo** [comment]
As long as you both agree this PR is good to go it’s just a formality who clicks the approve button 😃

I plan to have a closer look at this PR on Monday but you don’t need to wait for me.

**Comment by cbravobernal** [comment]
> As I was the author of the original PR, I guess it wouldn't be OK to approve and merge my own PR 😅 , although from what @gziolo said in [#36577 (comment)](https://github.com/WordPress/gutenberg/pull/36577#issuecomment-972767776) the PR should be ready to go.

I created the PR, I cannot assign the review to me, neither merge 😆 

I think the PR is good to go, after the merge, it will be updated once we merge and test the pagination links 😄

File: packages/block-library/src/comments-pagination/block.json
Comment (gziolo) [inline packages/block-library/src/comments-pagination/block.json]: A small thing:

```suggestion
	"title": "Comments Pagination",
```

File: packages/block-library/src/comments-pagination/edit.js
Comment (gziolo) [inline packages/block-library/src/comments-pagination/edit.js:44]: Names of those blocks need to be updated later.

File: packages/block-library/src/comments-pagination/index.php
Comment (gziolo) [inline packages/block-library/src/comments-pagination/index.php:17]: That’s a bit surprising but I checked the history of the Query Pagination block and it looks like they had an issue with rendering of an empty `div` when there was no pagination to show. Basically it bails out early when inner blocks render nothing.

File: packages/block-library/src/comments-pagination/editor.scss
Comment (gziolo) [inline packages/block-library/src/comments-pagination/editor.scss]: Where is it coming from? The class name isn’t prefixed with `wp-` so it raises a flag, but it might be as well some legacy name we need to maintain.

**Review by gziolo — APPROVED** [review]
Great work. Let’s land it and iterate in the other open PRs that introduce all necessary inner blocks. I left some feedback to consider as we continue development. I haven’t tested this PR as it’s going to be heavily tested with further work.

File: packages/block-library/src/comments-pagination/edit.js
Comment (DAreRodz) [reply packages/block-library/src/comments-pagination/edit.js:44]: Sure! I guess the best way to address it is to do that in the PRs that implement the inner blocks.

File: packages/block-library/src/comments-pagination/editor.scss
Comment (DAreRodz) [reply packages/block-library/src/comments-pagination/editor.scss]: That CSS class is the same that the [Query Pagination block has](https://github.com/WordPress/gutenberg/blob/6f3eb95d6f5eef0ff5a21ea5b2cf85ffba124e3d/packages/block-library/src/query-pagination/editor.scss#L17) ― just changing `query-pagination` by `comments-pagination` (potential bad copy-paste? 👀 ).

By the way, I realized that the class name `block-library-query-toolbar__popover` is being used inside the [`QueryToolbar` component](https://github.com/WordPress/gutenberg/blob/bbaeea2056c96af931170da1665d5d98c79b4b8d/packages/block-library/src/query/edit/query-toolbar.js#L47) (in the Query block), but the equivalent component for the Comments Query Loop block ― i.e. [`CommentsQueryLoopToolbar`](https://github.com/WordPress/gutenberg/blob/d51ac8a86b1756da64840b02f2ab5c129443391a/packages/block-library/src/comments-query-loop/toolbar.js) ― doesn't use any class.

I'm looking at whether it makes sense to add that style or not.

Comment (DAreRodz) [reply packages/block-library/src/comments-pagination/editor.scss]: OK, after taking a deeper look I don't see any reason to keep that style.

In fact, it seems we could also safely remove the [same code](https://github.com/WordPress/gutenberg/blob/6f3eb95d6f5eef0ff5a21ea5b2cf85ffba124e3d/packages/block-library/src/query-pagination/editor.scss#L17-L19) from the Query Pagination block as well, right? There is no element using that class (if I'm not mistaken).

That class actually looks like a copy of [this one](https://github.com/WordPress/gutenberg/blob/6244cf57555665dd55d64caad98674848f73678a/packages/block-library/src/query/editor.scss#L1-L3) from the Query block, which styles the [`QueryToolbar` component](https://github.com/WordPress/gutenberg/blob/bbaeea2056c96af931170da1665d5d98c79b4b8d/packages/block-library/src/query/edit/query-toolbar.js#L47) I mentioned previously.

**Comment by DAreRodz** [comment]
All tests were passing, so I have landed this PR. 🚀

**Pull request merged**
